app/controllers/application_controller.rb

```diff
@@ -5,36 +5,28 @@ class ApplicationController < ActionController::Base
   before_action :configure_permitted_parameters, if: :devise_controller?
   after_filter :store_location
 
-def store_location
-  # store last url - this is needed for post-login redirect to whatever the user last visited.
-  return unless request.get? 
-  if (request.path != "/users/sign_in" &&
-      request.path != "/users/sign_up" &&
-      request.path != "/users/password/new" &&
-      request.path != "/users/password/edit" &&
-      request.path != "/users/confirmation" &&
-      request.path != "/companies/create/confirmation" &&
-      request.path != "/companies/create" &&
-      request.path != "/companyInvites/authentication" &&
-      request.path != "/companyInvites/verification" &&
-      request.path != "/companies/create" &&
-      request.path != "/companyInvites/verification/confirmation" &&
-      !request.xhr?) # don't store ajax calls
-    session[:previous_url] = request.fullpath 
+  def store_location
+    # store last url - this is needed for post-login redirect to whatever the user last visited.
+    return unless request.get? 
+    if (request.path != "/users/sign_in" &&
+        request.path != "/users/sign_up" &&
+        request.path != "/users/password/new" &&
+        request.path != "/users/password/edit" &&
+        request.path != "/users/confirmation" &&
+        request.path != "/companies/create/confirmation" &&
+        request.path != "/companies/create" &&
+        request.path != "/companyInvites/authentication" &&
+        request.path != "/companyInvites/verification" &&
+        request.path != "/companies/create" &&
+        request.path != "/companyInvites/verification/confirmation" &&
+        !request.xhr?) # don't store ajax calls
+      session[:previous_url] = request.fullpath 
+    end
   end
-end
-
-def after_sign_in_path_for(resource)
-  session[:previous_url] || root_path
-end
-
-protected
 
-def after_update_path_for(resource)
-  @company = Company.find_by_id(current_user.company_id)
-  session[:company_username] = @company.username
-  user_path(resource)
-end
+  def after_sign_in_path_for(resource)
```

Comment:
Unused method argument - `resource`. If it's necessary, use `_` or `_resource` as an argument name to indicate that it won't be used. You can also write as `after_sign_in_path_for(*)` if you want the method to accept any arguments but don't care about them.